vertx-web/src/main/java/io/vertx/ext/web/handler/impl/BodyHandlerImpl.java

```diff
@@ -97,6 +97,12 @@ private BHandler(RoutingContext context) {
       });
     }
 
+    private void makeUploadDir(FileSystem fileSystem) {
+      if (!fileSystem.existsBlocking(uploadsDir)) {
```

Comment:
it should check if it's a directory. On a file (not a dir), the creation should fail